sim/common/item_effects.go

```diff
@@ -3507,6 +3508,69 @@ func init() {
 		})
 	})
 
+	// https://www.wowhead.com/classic/item=21670/badge-of-the-swarmguard
+	// Use: Chance on melee or ranged attack to grant 200 armor pen stacks up to 6 times.  Lasts 30 sec. (3 Min Cooldown)
+	core.NewItemEffect(BadgeOfTheSwarmguard, func(agent core.Agent) {
+		character := agent.GetCharacter()
+
+		procAura := character.RegisterAura(core.Aura{
+			Label: "Insight of the Qiraji",
+			ActionID: core.ActionID{SpellID: 26481},
+			Duration:  core.NeverExpires,
+			MaxStacks: 6,
+			OnStacksChange: func(aura *core.Aura, sim *core.Simulation, oldStacks int32, newStacks int32) {
+				character.AddStatDynamic(sim, stats.ArmorPenetration, 200*float64(newStacks-oldStacks))
+			},
+		})
+
+		ppmm := character.AutoAttacks.NewPPMManager(10.0, core.ProcMaskMeleeOrRanged)
+
+		auraLabel := "Badge of the Swarmguard"
+		actionID := core.ActionID{SpellID: 26480}
+		trinketAura := character.RegisterAura(core.Aura{
+			Label: auraLabel,
+			ActionID: actionID,
+			Duration: time.Second * 30,
+			OnExpire: func(aura *core.Aura, sim *core.Simulation) {
+				procAura.Deactivate(sim)
+			},
+			OnSpellHitDealt: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+				if !result.Landed() {
+					return
+				}
+				if !spell.ProcMask.Matches(core.ProcMaskMeleeOrRanged) {
+					return
+				}
+				if !ppmm.ProcWithWeaponSpecials(sim, spell.ProcMask, auraLabel) {
+					return
+				}
+
+				procAura.Activate(sim)
+				procAura.AddStack(sim)
+			},
+		})
```

Comment:
We can simplify this by using the `MakeProcTriggerAura` helper

```suggestion
		trinketAura := core.MakeProcTriggerAura(&character.Unit, core.ProcTrigger{
			Name: auraLabel,
			ActionID: actionID,
			Duration: time.Second * 30,
			Callback: core.CallbackOnSpellHitDealt,
			Outcome: core.OutcomeLanded,
			ProcMask: core.ProcMaskMeleeOrRanged,
			SpellFlagsExclude: core.SpellFlagSuppressEquipProcs,
			PPM: 10.0,
			Handler: func(sim *core.Simulation, _ *core.Spell, _ *core.SpellResult) {
				procAura.Activate(sim)
				procAura.AddStack(sim)
			},
		})
```